test/specs/trailApexReplayDebugger.e2e.ts

```diff
@@ -0,0 +1,228 @@
+    await testSetup.setUp();
+
+    // Create Apex class AccountService
+    await utilities.createApexClassWithBugs();
+
+    // Push source to org
+    const workbench = await browser.getWorkbench();
+    await utilities.runCommandFromCommandPrompt(workbench, 'SFDX: Push Source to Default Org and Override Conflicts', 1);
+    // Wait for the command to execute
+    await utilities.waitForNotificationToGoAway(workbench, 'Running SFDX: Push Source to Default Org and Override Conflicts', fiveMinutes);
+    const successPushNotificationWasFound = await utilities.notificationIsPresent(workbench, 'SFDX: Push Source to Default Org and Override Conflicts successfully ran');
+    expect(successPushNotificationWasFound).toBe(true);
+  });
+
+  step('Run Apex Tests', async () => {
+    // Run SFDX: Run Apex tests.
+    const workbench = await browser.getWorkbench();
+    await utilities.runCommandFromCommandPrompt(workbench, 'View: Clear Output', 1);
+    prompt = await utilities.runCommandFromCommandPrompt(workbench, 'SFDX: Run Apex Tests', 1);
+
+    // Select the "AccountServiceTest" file
+    await prompt.selectQuickPick('AccountServiceTest');
+
+    // Wait for the command to execute
+    await utilities.waitForNotificationToGoAway(workbench, 'Running SFDX: Run Apex Tests', fiveMinutes);
+    await utilities.waitForNotificationToGoAway(workbench, 'Running SFDX: Run Apex Tests: Listening for streaming state changes...', fiveMinutes);
+    await utilities.waitForNotificationToGoAway(workbench, 'Running SFDX: Run Apex Tests: Processing test run', fiveMinutes, false);
+
+    const successNotificationWasFound = await utilities.notificationIsPresent(workbench, 'SFDX: Run Apex Tests successfully ran');
+    expect(successNotificationWasFound).toBe(true);
+
+    // Verify test results are listed on vscode's Output section
+    const outputPanelText = await utilities.attemptToFindOutputPanelText('Apex', '=== Test Results', 10);
+    expect(outputPanelText).not.toBeUndefined();
+    expect(outputPanelText).toContain('Assertion Failed: incorrect ticker symbol');
+    expect(outputPanelText).toContain('Expected: CRM, Actual: SFDC');
+  });
+
+  step('Set Breakpoints and Checkpoints', async () => {
+    // Get open text editor
+    const workbench = await browser.getWorkbench();
+    const editorView = workbench.getEditorView();
+    const textEditor = await editorView.openEditor('AccountService.cls') as TextEditor;
+    await textEditor.moveCursor(8, 5);
+
+    // Run SFDX: Toggle Checkpoint.
+    prompt = await utilities.runCommandFromCommandPrompt(workbench, 'SFDX: Toggle Checkpoint', 1);
+
+    // Calling SFDX: Update Checkpoints in Org fails if the org has been recently created, 
+    // it does not complete the 6 steps but only 4.
+    // Reloading the window forces the extensions to be reloaded and this seems to fix the issue.
+    await utilities.runCommandFromCommandPrompt(workbench, 'Developer: Reload Window', 10);
+    await utilities.pause(15);
```

Comment:
OK, after chatting with you earlier, I spent 3 hours looking it this.  Here is what I found:
* I changed line 82 to 15 seconds, and line 89 to 5 seconds.
This ran successfully once
The second time it ran it failed (same error we were seeing before)

* I then changed line 82 back to 10, and line 89 to 10
This ran successfully once
The second time it ran it failed, but this time on line 191

* I change 191 from 1 to 2
This ran successfully three times
Then  on the fourth time, it failed at line 85 (not 89, but 85)

* I then changed line 82 from 10 to 15
This ran successfully once
The second time it ran it failed on line 85 again

* I then changed line 82 from 10 to 20
This ran successfully once
The second time it ran it failed on line 85 again
...and this ran successfully 6 times and I called it good.

That might have been hard to follow, so here's a summary of changes needed to work on my machine:
line 82: pause of 20
line 89: pause of 10
line 191: pause of 2